scripts/provisioning/s3setup/s3setup/cortx_s3_setup.py

```diff
@@ -0,0 +1,154 @@
+    log.debug("\ncmd:{0},\noutput:{1},\nerror:{2}".format(cmd, output, error))
+    if not error:
+      print ("Deleted s3backgrounddelete account successfully...")
+      return True 
+    print ("Delete s3backgrounddelete account failed with: {}".format(error))
+    return False
+
+  def delete_recovery_tool_account(self, ldappasswd: str, keylen: int, key: str, s3background_cofig:str):
+    """Delete s3 account which was used by s3recoverytool"""
+    cmd = 'ldapsearch -b "o=s3-recovery-svc,ou=accounts,dc=s3,dc=seagate,dc=com" -x -w ' + ldappasswd + ' -D "cn=sgiamadmin,dc=seagate,dc=com" -H ldap://'
+    output,error = subprocess.Popen(cmd, universal_newlines=True, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE).communicate()
+    log.debug("\ncmd:{0},\noutput:{1},\nerror:{2}".format(cmd, output, error))
+    output = re.sub(r"[\n\t\s]*","",output)
+    if "result:32Nosuchobjectmatched" in output:
+      print("No s3-recovery-svc account found")
+      return False
+    """Delete s3recovery tool account"""
+    s3_cipher = CortxS3Cipher(None, True, keylen, key)
+    access_key = ""
+    try:
+      access_key = s3_cipher.get_key()
+    except CipherInvalidToken as err:
+      log.debug("Cipher generate key failed with error : {0}, trying from flat file : {1}".format(err, s3background_cofig))
+      cmd = "awk '/background_account_access_key/ {print}' "+ s3background_cofig + " | cut -d " " -f 5 | sed -e 's/^\"//' -e 's/\"$//"
+      access_key,error = subprocess.Popen(cmd, universal_newlines=True, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE).communicate()
+      log.debug("\ncmd:{0},\noutput:{1},\nerror:{2}".format(cmd, access_key, error))
+    cmd = 'ldapdelete -x -w ' + ldappasswd + ' -r "ak=' + access_key + ',ou=accesskeys,dc=s3,dc=seagate,dc=com" -D "cn=sgiamadmin,dc=seagate,dc=com" -H ldapi:///'
+    output,error = subprocess.Popen(cmd, universal_newlines=True, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE).communicate()
+    log.debug("\ncmd:{0},\noutput:{1},\nerror:{2}".format(cmd, output, error))
+    cmd = 'ldapdelete -x -w ' + ldappasswd + ' -r "o=s3-recovery-svc,ou=accounts,dc=s3,dc=seagate,dc=com" -D "cn=sgiamadmin,dc=seagate,dc=com" -H ldapi:///'
+    output,error = subprocess.Popen(cmd, universal_newlines=True, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE).communicate()
+    log.debug("\ncmd:{0},\noutput:{1},\nerror:{2}".format(cmd, output, error))
+    if not error:
+      print ("Deleted s3recovery tool account successfully...")
+      return True 
+    print ("Delete s3recovery tool account failed with: {}".format(error))
+    return False
+
+  def accounts_cleanup(self, ldappasswd: str = None, s3background_cofig:str = "/opt/seagate/cortx/s3/s3backgrounddelete/config.yaml"):
+    """Clean up s3 accounts"""
+    if ldappasswd == None:
+      S3Cipher = CortxS3Cipher(None, True, 12, "openldap")
+      ldappasswd = S3Cipher.get_key()
```

Comment:
Can we make the ldappassword as mandatory option, and pass this from s3_setup script ?

As the get_key() function return cipher key, and that will no longer be the ldappassword.

Reply:
Ok, this will require some changes in the scripts, and need further testing. Can this be addressed in a followup ticket

Reply:
I've addressed this change after picking your latest changes from mini-provisioner head. Please review.

Reply:
this section of code is not needed now, remove it in next PR